src/Provider/HttpKernelServiceProvider.php

```diff
@@ -61,6 +58,21 @@ public function register(\Pimple\Container $app)
                 $app['phpdi.callable_resolver']
             );
         };
+
+        // Override the argument resolver with ours.
+        $app['argument_resolver'] = function () {
+            return new ArgumentResolver(
+                new ResolverChain([
+                    new AssociativeArrayResolver,
+                    new TypeHintContainerResolver($this->container),
+                ])
+            );
+        };
+
+        // add request to ensure that DI have access to correct request instance from application
+        $app['request'] = $app->factory(function ($app) {
+            return $app['request_stack']->getCurrentRequest();
+        });
```

Comment:
Since the container should be stateless (and all the registered services too) the request should not be stored in the container. Instead services should inject the request stack and resolve the request when they need it.

Here the request will be resolved once and stored by the container (so the same object will be returned every time).

Reply:
Request is assign by factory method so `request` object will be instance which currently is used in Application. Every call will return current request instance.

In few places in application I'm using `Request` object. Without that I'm getting new instance and should be instance used by application with filled arguments. I don't want to change it to `RequestStack` in every occurrence. So this solution seams valid: if you have `Request` object to inject in other places than controller you want to receive original `Request` object not new one.

Reply:
Ah you're right I thought it was a "PHP-DI factory", which is not resolved every time.

Anyway there should be only stateless objects in the container ([here is a good intro](https://igor.io/2013/03/31/stateless-services.html) on the subject). You can register that binding in your projects though if you want it that way.

Reply:
Ok, I can remove it if you don't want to have it here. I will register that binding in my project.